app/core/api/api_key_views.py

```diff
@@ -0,0 +1,61 @@
+# from rest_framework import serializers as rest_serializers
+from drf_spectacular.utils import extend_schema
+from drf_spectacular.utils import extend_schema_view
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework_api_key.permissions import HasAPIKey
+
+from core.api.serializers import PracticeAreaSerializer
+from core.api.serializers import UserSerializer
+from core.models import PracticeArea
+from core.models import User
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="""
+Lists all users and the associated groups for the user
+
+Requires
+- API keys have been added using the API key screen in the People depot App.
+Check with People Depot admin.
+- Requires X-Api-Key be set when submitting request
+For curl, it would look like this:
+```http
+curl -L -X GET "localhost:8001/api/v1/apikey/getusers/"
+-H "X-Api-Key: *************"
+-H "Content-Type: application/json"
+```
+
+
+For python, it would look like this:
+```bash
+import requests
+
+API_KEY = os.environ.get("API_KEY")
+BASE_URL = "http:8000"
+HEADERS = {
+    'X-Api-Key': API_KEY,
+    'Content-Type': 'application/json'
+}
+response = requests.get(f"{BASE_URL}/api/v1/apikey/getusers/", headers=HEADERS)
+```
+        """
+    )
+)
+class ApiKeyUserViewSet(viewsets.ReadOnlyModelViewSet):
+    # HasAPIKey checks if value of X-API-KEY is in API keys table
+    permission_classes = [IsAuthenticatedOrReadOnly, HasAPIKey]
+    queryset = PracticeArea.objects.all()
+    serializer_class = PracticeAreaSerializer  # HasAPIKey checks against keys stored in
+    queryset = User.objects.all()
+
+    # when instantiated, get_serializer_context will be called
+    serializer_class = UserSerializer
```

Comment:
This probably should have been settled in the planning stage of this feature rather than during the PR review.

To clarify further, the problem I see is this new API key authentication method allows the API client to view `user` data without any of the limitations that are applied to permission-tiered users.

These are the fields it returns for `user`:
```
        fields = (
            "uuid",
            "username",
            "created_at",
            "updated_at",
            "email",
            "first_name",
            "last_name",
            "gmail",
            "preferred_email",
            "current_job_title",
            "target_job_title",
            "current_skills",
            "target_skills",
            "linkedin_account",
            "github_handle",
            "slack_id",
            "phone",
            "texting_ok",
            "time_zone",
            "groups",
        )
```

I'm thinking that KnowledgeBase doesn't really need or care about a user's `phone` number to function.

So I think we need to consult the "permissions team" (@ExperimentsInHonesty @Neecolaa) to figure out/negotiate what data this should include based on what data KnowledgeBase requires to function.

This seems to mean adding more/duplicate work to the "permissions" team (@ExperimentsInHonesty and @Neecolaa) to define exactly what tables and fields this new method can allow an API client to access. Would it be better to say the API client has the same permissions as one of the already-defined user permissions tier such as `unverifiedUser` or `stakeholder API`? In that case, wouldn't it be simpler to just use a service account (special user account with the right permission level) instead?

For reference: here's the [Field Permissions sheet](https://docs.google.com/spreadsheets/d/1x_zZ8JLS2hO-zG0jUocOJmX16jh-DF5dccrd_OEGNZ0/edit?gid=371053454#gid=371053454) being worked on currently.